# Introduction

Ce jupyter notebook a pour but de :
- déterminer les vecteurs de bases des réseaux direct et réciproque,
- le type de maille,
- le système cristallin, 
- le groupe ponctuel

du matériau qui m'a été attribué; le **CsNdTe2** (mp-11742).

### (0) importation des modules nécessaires 

In [6]:
# On importe tout les modules qui nous seront potentiellement utiles
from pymatgen.ext.matproj import MPRester
from pymatgen.apps.borg.hive import VaspToComputedEntryDrone
from pymatgen.apps.borg.queen import BorgQueen
from pymatgen.entries.compatibility import MaterialsProjectCompatibility
from pymatgen.analysis.phase_diagram import PhaseDiagram, PDPlotter
from jupyter_jsmol.pymatgen import quick_view
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.core.lattice import Lattice

# On télécharge le fichier CIF du matériau sur le MP sur base de l'API
with MPRester("NwZmcoOXaAFq0K75qQZ") as m:

    # Structure for material id
    structure = m.get_structure_by_material_id("mp-11742")



### (1) Vecteurs de bases des réseaux direct et réciproque

In [35]:
# RD = réseau direct; RR = réseau réciproque
vectRD = structure.lattice
vectRR = structure.lattice.reciprocal_lattice

print("----------données brutes---------")
print ("---> coordonnées cartésiennes des vecteurs de bases (Réseau direct)")
print(vectRD)
print("---> norme des vecteurs a, b, et c (paramètres de maille)")
print(vectRD.abc)
print("---> angles entres les vecteurs de bases")
print(vectRD.angles)
print ("---> coordonnées cartésiennes des vecteurs de bases (Réseau réciproque)")
print(vectRR)
print("---> norme des vecteurs a*, b*, et c* (paramètres de maille)")
print(vectRR.abc)
print("---> angles entres les vecteurs de bases")
print(vectRR.angles)





----------données brutes---------
---> coordonnées cartésiennes des vecteurs de bases (Réseau direct)
9.054383 -2.341648 0.000000
9.054383 2.341648 0.000000
8.448785 0.000000 4.010386
---> norme des vecteurs a, b, et c (paramètres de maille)
(9.352281372295906, 9.352281372295906, 9.352281210764623)
---> angles entres les vecteurs de bases
(29.000300008372854, 29.000300008372854, 29.00030060443356)
---> coordonnées cartésiennes des vecteurs de bases (Réseau réciproque)
0.346969 -1.341616 -0.730969
0.346969 1.341616 -0.730969
0.000000 0.000000 1.566728
---> norme des vecteurs a*, b*, et c* (paramètres de maille)
(1.5667282602563706, 1.5667282602563706, 1.566728316720532)
---> angles entres les vecteurs de bases
(117.81108428138728, 117.81108428138728, 117.81108224283874)


Les coordonnées cartésiennes des vecteurs de bases du réseau directe sont:
- $\overrightarrow{a} = (9.054383, -2.341648, 0.000000)$ et $ \| \overrightarrow{a} \| = 9.352281372295906   [angstrom] $  
- $\overrightarrow{b} = (9.054383, 2.341648, 0.000000)$ et $ \| \overrightarrow{b} \| = 9.352281372295906  $
- $\overrightarrow{c} = (8.448785, 0.000000, 4.010386)$ et $ \| \overrightarrow{c} \| = 9.352281372295906  $

Les angles formés par les vecteurs de base du réseau direct sont:
- $\alpha = 29.000300008372854 ^{\circ} $ 
- $\beta = 29.000300008372854 ^{\circ} $
- $\gamma = 29.000300008372854 ^{\circ}$

Les coordonnées cartésiennes des vecteurs de bases du réseau directe sont:
- $\overrightarrow{a*} = (0.346969, -1.341616, -0.730969)$ et $ \| \overrightarrow{a*} \| = 1.5667282602563706 [angstrom]  $ 
- $\overrightarrow{b*} = (0.346969, 1.341616, -0.730969)$ et $ \| \overrightarrow{b*} \| = 1.5667282602563706  $ 
- $\overrightarrow{c*} = (0.000000, 0.000000, 1.566728)$ et $ \| \overrightarrow{c*} \| = 1.5667282602563706  $ 

Les angles formés par les vecteurs de base du réseau réciproque sont:
- $\alpha* = 117.81108428138728 ^{\circ} $ 
- $\beta* = 117.81108428138728 ^{\circ} $
- $\gamma* = 117.81108428138728 ^{\circ}$


### (2) Type de maille 

In [22]:
type_de_maille = SpacegroupAnalyzer(structure).get_lattice_type()
print("type de maille:", type_de_maille)

type de maille: rhombohedral


La maille primitive du matériau est **rhombohedrale**.

### (3) Système cristallin 

In [41]:
systeme_cristallin = SpacegroupAnalyzer(structure).get_crystal_system()
print("système cristallin:",systeme_cristallin)

système cristallin: trigonal


Le système cristallin du matériau est **trigonal**.

### (4) Groupe ponctuel

In [15]:
groupeponctuel = SpacegroupAnalyzer(structure).get_point_group_symbol()
groupedespace = SpacegroupAnalyzer(structure).get_space_group_symbol()
print("groupe ponctuel:",groupeponctuel)
print("groupe d'espace :", groupedespace)

groupe ponctuel: -3m
groupe d'espace : R-3m


Le groupe ponctuel du matériau est **-3m** et son groupe d'espace est **R-3m** (166). 

### (5) Conclusion

En conclusion, on visualise la cellule primitive du matériau. Les caractéristiques de celle-ci, affichée en haut à gauche concordent avec les résultats obtenus précedemment. 

In [11]:
view = quick_view(structure)
display(view) 

JsmolView(layout=Layout(align_self='stretch', height='400px'))